src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java

```diff
@@ -60,6 +61,17 @@ public void testOverloadMethodsDeclarationOrderRecords() throws Exception {
             expected);
     }
 
+    @Test
+    public void testOverloadMethodsDeclarationOrderPrivateAndStaticMethods() throws Exception {
```

Comment:
in addition please add this test to
https://github.com/checkstyle/checkstyle/blob/master/src/it/java/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/OverloadMethodsDeclarationOrderTest.java

to make it firm that no matter how this Check might change in design, google style config works good on such code snippet.

Reply:
done